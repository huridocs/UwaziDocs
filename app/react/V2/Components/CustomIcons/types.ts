interface CustomIconProps {
  className?: string;
}

export type { CustomIconProps };
