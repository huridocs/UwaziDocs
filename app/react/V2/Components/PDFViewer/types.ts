import { TextSelection } from 'react-text-selection-handler/dist/TextSelection';

type TextHighlight = {
  key: string;
  textSelection: TextSelection;
  color?: string;
};

export type { TextHighlight };
