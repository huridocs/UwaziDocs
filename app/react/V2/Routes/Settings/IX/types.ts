import { IXExtractorInfo } from 'V2/shared/types';
import { PropertySchema } from 'shared/types/commonTypes';

interface IXProperty extends PropertySchema {
  type: 'text' | 'date' | 'numeric' | 'markdown';
}

type Extractor = IXExtractorInfo & {
  namedTemplates: string[];
  propertyLabel: string;
  propertyType: IXProperty['type'];
};

export type { Extractor };
