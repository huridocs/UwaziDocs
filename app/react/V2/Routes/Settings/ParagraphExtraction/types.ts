export type ParagraphExtractorValues = {
  _id?: string;
  templatesFrom: string[];
  templateTo: string;
};

export type Extractor = ParagraphExtractorValues & {
  documents: number;
  generatedEntities: number;
};

export type TableExtractor = Extractor & {
  rowId: string;
  targetTemplateName: string;
  originTemplateNames: string[];
};
