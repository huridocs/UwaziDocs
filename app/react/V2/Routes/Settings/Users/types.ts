type FormIntent =
  | 'new-user'
  | 'edit-user'
  | 'delete-users'
  | 'new-group'
  | 'edit-group'
  | 'delete-groups'
  | 'unlock-user'
  | 'reset-password'
  | 'reset-2fa'
  | 'bulk-reset-2fa'
  | 'bulk-reset-password';

export type { FormIntent };
