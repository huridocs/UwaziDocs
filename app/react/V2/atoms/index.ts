export { atomStore } from './store';
export { notificationAtom } from './notificationAtom';
export { settingsAtom } from './settingsAtom';
export { templatesAtom } from './templatesAtom';
export { translationsAtom } from './translationsAtom';
export { thesaurisAtom } from './thesaurisAtom';
export { globalMatomoAtom } from './globalMatomoAtom';
export { userAtom } from './userAtom';
export type { AtomStoreData } from './store';
export type { notificationAtomType } from './notificationAtom';
