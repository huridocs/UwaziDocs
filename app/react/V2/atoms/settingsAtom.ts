import { atom } from 'recoil';
import { ClientSettings } from 'app/apiResponseTypes';

const settingsAtom = atom({
  key: 'settings',
  default: {} as ClientSettings,
});

export { settingsAtom };
