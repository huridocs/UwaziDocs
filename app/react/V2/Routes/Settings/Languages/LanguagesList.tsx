/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import { IncomingHttpHeaders } from 'http';
import { useLoaderData, LoaderFunction } from 'react-router-dom';
import { useRecoilValue } from 'recoil';
import { intersectionBy, keyBy, merge, values } from 'lodash';
import { ColumnDef, Row, createColumnHelper } from '@tanstack/react-table';
import { Translate, I18NApi } from 'app/I18N';
import { RequestParams } from 'app/utils/RequestParams';
import { settingsAtom } from 'app/V2/atoms/settingsAtom';
import { Button, Table, ConfirmationModal } from 'V2/Components/UI';
import { useApiCaller } from 'V2/CustomHooks/useApiCaller';
import { SettingsContent } from 'app/V2/Components/Layouts/SettingsContent';
import { LanguageSchema } from 'shared/types/commonTypes';
import { ClientSettings } from 'app/apiResponseTypes';
import { InstallLanguagesModal } from './components/InstallLanguagesModal';
import {
  DefaultHeader,
  LabelHeader,
  ResetHeader,
  UninstallHeader,
  DefaultButton,
  ResetButton,
  UninstallButton,
  LanguageLabel,
} from './components/TableComponents';

const languagesListLoader =
  (headers?: IncomingHttpHeaders): LoaderFunction =>
  async () =>
    I18NApi.getLanguages(new RequestParams({}, headers));
// eslint-disable-next-line max-statements
const LanguagesList = () => {
  const { languages: collectionLanguages = [] } = useRecoilValue<ClientSettings>(settingsAtom);
  const { requestAction } = useApiCaller();
  const [modalProps, setModalProps] = useState({});
  const [showModal, setShowModal] = useState(false);
  const [showInstallModal, setShowInstallModal] = useState(false);

  const availableLanguages = useLoaderData() as LanguageSchema[];
  const installedLanguages = intersectionBy(availableLanguages, collectionLanguages, 'key');
  const notInstalledLanguages = availableLanguages.filter(
    l => !collectionLanguages.find(cl => cl.key === l.key)
  );
  const languages = values(
    merge(keyBy(installedLanguages, 'key'), keyBy(collectionLanguages, 'key'))
  );

  const handleAction =
    (
      successMessage: React.ReactNode,
      action: { (requestParams: RequestParams): Promise<Response> },
      key: string,
      currentLanguage?: LanguageSchema
    ) =>
    async () => {
      setShowModal(false);
      if (currentLanguage) {
        await requestAction(
          action,
          new RequestParams({ [key]: currentLanguage.key }),
          successMessage
        );
      }
    };

  const confirmAction = (
    message: string,
    acceptLabel: string,
    handleAcceptedAction: () => void
  ) => {
    setModalProps({
      header: 'Are you sure?',
      body: message,
      acceptButton: acceptLabel,
      cancelButton: 'No, cancel',
      warningText: 'Other users will be affected by this action!',
      confirmWord: 'CONFIRM',
      onAcceptClick: handleAcceptedAction,
      onCancelClick: () => setShowModal(false),
      size: 'md',
    });
    setShowModal(true);
  };

  const resetModal = (row: Row<LanguageSchema>) => {
    confirmAction(
      'You are about to reset a language.',
      'Reset',
      handleAction(
        <Translate>Language reset success</Translate>,
        I18NApi.populateTranslations,
        'locale',
        row.original as LanguageSchema
      )
    );
  };

  const setDefaultLanguage = async (row: Row<LanguageSchema>) => {
    await handleAction(
      <Translate>Default language change success</Translate>,
      I18NApi.setDefaultLanguage,
      'key',
      row.original as LanguageSchema
    )();
  };

  const uninstallModal = (row: Row<LanguageSchema>) => {
    confirmAction(
      'You are about to uninstall a language.',
      'Uninstall',
      handleAction(
        <Translate translationKey="Language Uninstall Start Message">
          Language uninstallation process initiated. It may take several minutes to complete
          depending on the size of the collection. Please wait until the uninstallation process is
          finished.
        </Translate>,
        I18NApi.deleteLanguage,
        'key',
        row.original as LanguageSchema
      )
    );
  };

  // Helper typed as any because of https://github.com/TanStack/table/issues/4224
  const columnHelper = createColumnHelper<any>();
  const columns = [
    columnHelper.accessor('label', {
      id: 'label',
      header: LabelHeader,
      cell: LanguageLabel,
      meta: { headerClassName: 'w-9/12' },
    }) as ColumnDef<LanguageSchema, 'label'>,
    columnHelper.accessor('default', {
      header: DefaultHeader,
      cell: DefaultButton,
      enableSorting: false,
      meta: { action: setDefaultLanguage, headerClassName: 'text-center w-1/12' },
    }) as ColumnDef<LanguageSchema, 'default'>,
    columnHelper.accessor('key', {
      header: ResetHeader,
      cell: ResetButton,
      enableSorting: false,
      meta: { action: resetModal, headerClassName: 'text-center w-1/12' },
    }) as ColumnDef<LanguageSchema, 'key'>,
    columnHelper.accessor('_id', {
      header: UninstallHeader,
      cell: UninstallButton,
      enableSorting: false,
      meta: { action: uninstallModal, headerClassName: 'text-center w-1/12' },
    }) as ColumnDef<LanguageSchema, '_id'>,
  ];

  return (
    <div
      className="tw-content"
      style={{ width: '100%', overflowY: 'auto' }}
      data-testid="settings-languages"
    >
      <SettingsContent>
        <SettingsContent.Header title="Languages" />
        <SettingsContent.Body>
          <div data-testid="languages">
            <Table<LanguageSchema>
              columns={columns}
              data={languages}
              title={<Translate>Active languages</Translate>}
              initialState={{ sorting: [{ id: 'label', desc: false }] }}
            />
          </div>
        </SettingsContent.Body>
        <SettingsContent.Footer>
          <div className="flex gap-2">
            <Button
              onClick={() => {
                setShowInstallModal(true);
              }}
            >
              <Translate>Install Language(s)</Translate>
            </Button>
          </div>
        </SettingsContent.Footer>
      </SettingsContent>
      {showModal && (
        <div className="container w-10 h-10">
          <ConfirmationModal {...modalProps} size="md" />
        </div>
      )}
      {showInstallModal && (
        <InstallLanguagesModal
          setShowModal={setShowInstallModal}
          languages={notInstalledLanguages}
        />
      )}
    </div>
  );
};

export { LanguagesList, languagesListLoader };
