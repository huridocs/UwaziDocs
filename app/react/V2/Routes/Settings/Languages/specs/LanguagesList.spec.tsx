/**
 * @jest-environment jsdom
 */
import React from 'react';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import { screen, RenderResult, fireEvent, within, act } from '@testing-library/react';
import { defaultState, renderConnectedContainer } from 'app/utils/test/renderConnected';
import { I18NApi } from 'app/I18N';
import { settingsAtom } from 'app/V2/atoms/settingsAtom';
import { Settings } from 'shared/types/settingsType';
import * as useApiCaller from 'app/V2/CustomHooks/useApiCaller';
import { LanguagesList } from '../LanguagesList';

const languageDefinition = (
  label: string,
  key: string,
  ISO639_3: string,
  // eslint-disable-next-line camelcase
  localized_label: string,
  translationAvailable: boolean = false
  // eslint-disable-next-line max-params
) => ({
  label,
  key,
  ISO639_3,
  // eslint-disable-next-line camelcase
  localized_label,
  translationAvailable,
});
const abkhazianLanguage = languageDefinition('Abkhazian', 'ab', 'abk', 'Abkhazian');
const availableLanguages = [
  { ...abkhazianLanguage },
  { ...languageDefinition('English', 'en', 'eng', 'English') },
  { ...languageDefinition('Spanish', 'es', 'spa', 'Spanish', true) },
  { ...languageDefinition('Afar', 'aa', 'aar', 'Afar') },
  { ...languageDefinition('Thai', 'th', 'tha', 'ไทย', true) },
];

const currentLanguages = [
  { label: 'English', key: 'en' },
  { label: 'Spanish', key: 'es', default: true },
  { label: 'Arabic', key: 'ar' },
];
const mockUseLoaderData = jest.fn().mockImplementation(() => availableLanguages);

jest.mock('react-router-dom', () => ({
  useLoaderData: () => mockUseLoaderData(),
  // eslint-disable-next-line jsx-a11y/anchor-has-content, react/prop-types, react/jsx-props-no-spreading
  Link: (props: { to: string }) => <a {...props} href={props.to} />,
}));

describe('LanguagesList', () => {
  let renderResult: RenderResult;

  const recoilGlobalState = ({ set }: MutableSnapshot) => {
    const settings: Partial<Settings> = { languages: currentLanguages };
    set(settingsAtom, settings);
  };

  const render = () => {
    ({ renderResult } = renderConnectedContainer(
      <RecoilRoot initializeState={recoilGlobalState}>
        <LanguagesList />
      </RecoilRoot>,
      () => defaultState
    ));
  };
  let rows: HTMLElement[];

  const requestActionMock = jest.fn();

  beforeEach(() => {
    spyOn(I18NApi, 'setDefaultLanguage').and.callFake(async () => Promise.resolve({}));
    spyOn(useApiCaller, 'useApiCaller').and.callFake(() => ({
      requestAction: requestActionMock,
      data: undefined,
      error: '',
    }));
    render();
    rows = screen.getAllByRole('row');
  });

  describe('render installed languages', () => {
    it('should show a list of installed languages', () => {
      expect(rows[1].children[0].textContent).toEqual('Arabic (ar)');
      expect(rows[2].children[0].textContent).toEqual('English (en)');
      expect(rows[3].children[0].textContent).toEqual('Spanish (es)');

      expect(renderResult.container).toMatchSnapshot();
    });
    const checkButtonState = (row: number, column: number, status: string) => {
      expect(rows[row].children[column].getElementsByTagName('button')[0].className).toContain(
        status
      );
    };
    const checkButtonDefinition = (row: number, column: number, count: number) => {
      expect(rows[row].children[column].getElementsByTagName('button').length).toBe(count);
    };
    it('should highlight the default language', () => {
      checkButtonState(1, 1, 'disabled');
      checkButtonState(2, 1, 'disabled');
      checkButtonState(3, 1, 'enabled');
    });
    it('should allow reset a language if there are translations available', () => {
      checkButtonDefinition(1, 2, 0);
      checkButtonDefinition(2, 2, 0);
      checkButtonDefinition(3, 2, 1);
    });
    it('should allow uninstalling any language except default', () => {
      checkButtonDefinition(1, 3, 1);
      checkButtonDefinition(2, 3, 1);
      checkButtonDefinition(3, 3, 0);
    });
  });

  describe('actions', () => {
    it('should reset a language', async () => {
      await act(async () => {
        fireEvent.click(within(rows[3].children[2] as HTMLElement).getByRole('button'));
      });
      await act(async () => {
        fireEvent.change(screen.getByRole('textbox'), {
          target: { value: 'CONFIRM' },
        });
        fireEvent.click(within(screen.getByTestId('modal')).getByRole('button', { name: 'Reset' }));
        expect(requestActionMock).toHaveBeenCalledWith(
          I18NApi.populateTranslations,
          {
            data: { locale: 'es' },
            headers: {},
          },
          'Language reset success'
        );
      });
    });

    it('should set a language as default', async () => {
      await act(() => {
        fireEvent.click(rows[1].children[1].getElementsByTagName('button')[0]);
      });
      expect(requestActionMock).toHaveBeenCalledWith(
        I18NApi.setDefaultLanguage,
        {
          data: { key: 'ar' },
          headers: {},
        },
        'Default language change success'
      );
    });
    it('should allow to cancel an action', async () => {
      await act(async () => {
        fireEvent.click(within(rows[1].children[3] as HTMLElement).getByRole('button'));
      });
      await act(async () => {
        fireEvent.change(screen.getByRole('textbox'), {
          target: { value: 'CONFIRM' },
        });
        fireEvent.click(
          within(screen.getByTestId('modal')).getByRole('button', { name: 'No, cancel' })
        );
        expect(requestActionMock).not.toHaveBeenCalledWith();
      });
    });

    it('should uninstall a language', async () => {
      await act(async () => {
        fireEvent.click(within(rows[2].children[3] as HTMLElement).getByRole('button'));
      });
      await act(async () => {
        fireEvent.change(screen.getByRole('textbox'), {
          target: { value: 'CONFIRM' },
        });
        fireEvent.click(
          within(screen.getByTestId('modal')).getByRole('button', { name: 'Uninstall' })
        );
        expect(requestActionMock).toHaveBeenCalledWith(
          I18NApi.deleteLanguage,
          {
            data: { key: 'en' },
            headers: {},
          },
          'Language uninstalled success'
        );
      });
    });
  });
});
