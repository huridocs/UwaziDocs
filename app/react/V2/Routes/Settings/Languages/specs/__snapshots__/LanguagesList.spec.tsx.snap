// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagesList render installed languages should show a list of installed languages 1`] = `
<div>
  <div
    class="tw-content"
    data-testid="settings-languages"
    style="width: 100%; overflow-y: auto;"
  >
    <div
      class="flex flex-col h-full"
    >
      <div
        class="flex-grow p-5"
      >
        <div
          class="flex items-center"
        >
          <a
            class="block lg:hidden"
            href="/settings"
            to="/settings"
          >
            <div
              class="flex mr-3"
            >
              <svg
                aria-hidden="true"
                class="w-6 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.75 19.5L8.25 12l7.5-7.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span
                class="sr-only"
              >
                <span
                  class="translation active"
                >
                  Navigate back
                </span>
              </span>
            </div>
          </a>
          <h1
            class="text-base text-gray-700"
          >
            <span
              class="translation active"
            >
              Languages
            </span>
          </h1>
        </div>
        <div
          class="mt-4"
          data-testid="languages"
        >
          <div
            class="relative"
            data-testid="table-element"
          >
            <div
              class="absolute bg-white dark:bg-black w-full h-full top-0 left-0 rounded-lg drop-shadow-md -z-10"
            />
            <table
              class="w-full text-left text-sm text-gray-500 dark:text-gray-400"
              role="table"
            >
              <caption
                class="p-5 text-lg font-semibold text-left text-gray-900 bg-white"
              >
                <span
                  class="translation active"
                >
                  Active languages
                </span>
              </caption>
              <thead
                class="group/head text-xs uppercase text-gray-700 dark:text-gray-400"
              >
                <tr>
                  <th
                    class="group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg bg-gray-50 dark:bg-gray-700 px-6 py-3 w-9/12"
                    colspan="1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="text-gray-500 flex flex-row"
                    >
                      <span
                        class="translation active"
                      >
                        Language
                      </span>
                      <svg
                        aria-hidden="true"
                        class="w-4"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M14.77 12.79a.75.75 0 01-1.06-.02L10 8.832 6.29 12.77a.75.75 0 11-1.08-1.04l4.25-4.5a.75.75 0 011.08 0l4.25 4.5a.75.75 0 01-.02 1.06z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </th>
                  <th
                    class="group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg bg-gray-50 dark:bg-gray-700 px-6 py-3 text-center w-1/12"
                    colspan="1"
                    role="columnheader"
                  >
                    <div
                      class="text-gray-500 "
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Default language
                      </span>
                    </div>
                  </th>
                  <th
                    class="group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg bg-gray-50 dark:bg-gray-700 px-6 py-3 text-center w-1/12"
                    colspan="1"
                    role="columnheader"
                  >
                    <div
                      class="text-gray-500 "
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Reset language
                      </span>
                    </div>
                  </th>
                  <th
                    class="group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg bg-gray-50 dark:bg-gray-700 px-6 py-3 text-center w-1/12"
                    colspan="1"
                    role="columnheader"
                  >
                    <div
                      class="text-gray-500 "
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Uninstall language
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="group/body text-gray-900 divide-y"
                role="rowgroup"
              >
                <tr
                  class="group/row"
                  data-testid="table-row-element"
                  role="row"
                >
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <span
                      class="translation active"
                    >
                      Arabic (ar)
                    </span>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-gray-700 disabled:text-gray-300 bg-white border-gray-200 enabled:hover:text-primary-700 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Default
                      </span>
                      <svg
                        aria-hidden="true"
                        class=" w-5 text-white stroke-current stroke-gray-300 stroke-2"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                     
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-primary-700 bg-white border-primary-700 disabled:text-primary-300 disabled:border-primary-300 enabled:hover:bg-primary-800 enabled:hover:text-white enabled:hover:border-primary-800 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active"
                      >
                        Uninstall
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="group/row"
                  data-testid="table-row-element"
                  role="row"
                >
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <span
                      class="translation active"
                    >
                      English (en)
                    </span>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-gray-700 disabled:text-gray-300 bg-white border-gray-200 enabled:hover:text-primary-700 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Default
                      </span>
                      <svg
                        aria-hidden="true"
                        class=" w-5 text-white stroke-current stroke-gray-300 stroke-2"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                     
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-primary-700 bg-white border-primary-700 disabled:text-primary-300 disabled:border-primary-300 enabled:hover:bg-primary-800 enabled:hover:text-white enabled:hover:border-primary-800 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active"
                      >
                        Uninstall
                      </span>
                    </button>
                  </td>
                </tr>
                <tr
                  class="group/row"
                  data-testid="table-row-element"
                  role="row"
                >
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <span
                      class="translation active"
                    >
                      Spanish (es)
                    </span>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-white bg-primary-700 border-primary-700 disabled:bg-primary-300 disabled:border-primary-300 enabled:hover:bg-primary-800 enabled:hover:border-primary-800 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active sr-only"
                      >
                        Default
                      </span>
                      <svg
                        aria-hidden="true"
                        class="w-5"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                    <button
                      class="text-primary-700 bg-white border-primary-700 disabled:text-primary-300 disabled:border-primary-300 enabled:hover:bg-primary-800 enabled:hover:text-white enabled:hover:border-primary-800 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
                      type="button"
                    >
                      <span
                        class="translation active"
                      >
                        Reset
                      </span>
                    </button>
                  </td>
                  <td
                    class="group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg px-6 py-4"
                    role="cell"
                  >
                     
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="fixed bottom-0 left-0 w-full p-1 bg-white border-t border-gray-200 lg:sticky z-1"
      >
        <div
          class="flex gap-2 p-2 pt-1"
        >
          <button
            class="text-white bg-primary-700 border-primary-700 disabled:bg-primary-300 disabled:border-primary-300 enabled:hover:bg-primary-800 enabled:hover:border-primary-800 text-sm px-5 py-2.5 disabled:cursor-not-allowed font-medium rounded-lg
      border focus:outline-none focus:ring-4 focus:ring-indigo-200 "
            type="button"
          >
            <span
              class="translation active"
            >
              Install language(s)
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
